File: backend/internal/models/booking.go
package models

import "time"

type Booking struct {
	ID        uint      `gorm:"primaryKey" json:"id"`
	UserID    uint      `json:"user_id"`
	EventID   uint      `json:"event_id"`
	Quantity  int       `json:"quantity"`
	CreatedAt time.Time `json:"created_at"`

	Status        string `json:"status"`
	PaymentStatus string `json:"payment_status"`
	OrderID       string `json:"order_id"`
	PaymentID     string `json:"payment_id"`
	Amount        int    `json:"amount"`

	User  User  `gorm:"foreignKey:UserID"`
	Event Event `gorm:"foreignKey:EventID"`
}

File: frontend/types/booking.ts
import { Event, Seat } from "./event"

export interface Booking {
  id: number
  user_id: number
  event_id: number
  quantity: number
  status: string
  payment_status: string
  order_id: string
  payment_id: string
  razorpay_order_id: string
  amount: number
  created_at: string
  event: Event
  seats: Seat[]
}